src/data_source/storage/sql/queries/state.rs

```diff
@@ -321,13 +342,37 @@ pub(crate) fn build_hash_batch_insert(
 // Represents a row in a state table
 #[derive(Debug, Default, Clone, FromRow)]
 pub(crate) struct Node {
-    pub(crate) path: Vec<i32>,
+    pub(crate) path: JsonValue,
     pub(crate) created: i64,
     pub(crate) hash_id: i32,
-    pub(crate) children: Option<Vec<i32>>,
-    pub(crate) children_bitvec: Option<BitVec>,
-    pub(crate) index: Option<serde_json::Value>,
-    pub(crate) entry: Option<serde_json::Value>,
+    pub(crate) children: Option<JsonValue>,
+    pub(crate) children_bitvec: Option<Vec<u8>>,
+    pub(crate) idx: Option<JsonValue>,
+    pub(crate) entry: Option<JsonValue>,
+}
+
+impl<'a, R> From<&'a R> for Node
+where
+    R: Row,
+    JsonValue: 'a + Decode<'a, <R as Row>::Database>,
+    Option<JsonValue>: Decode<'a, <R as Row>::Database>,
+    Vec<u8>: Decode<'a, <R as Row>::Database>,
+    i64: Decode<'a, <R as Row>::Database>,
+    i32: Decode<'a, <R as Row>::Database>,
+
+    &'a str: ColumnIndex<R>,
+{
+    fn from(row: &'a R) -> Self {
```

Comment:
Why did we add this? Doesn't `FromRow` still work?

Reply:
yeah the FromRow does type checking which is broken for sqlite

Reply:
Ah right. We should not derive `FromRow` then if it doesn't work